src/test/java/com/datalogics/pdf/samples/printing/PrintPdfTest.java

```diff
@@ -111,4 +142,58 @@ private void savePageImage(final BufferedImage image, final int pageIndex) throw
             }
         }
     }
+
+    /*
+     * TestMultiPagePrinterJob implements a 'fake' PrinterJob to intercept print requests.
+     *
+     * Note: This class will be used to test printing a page multiple times. This will test the scenario when printer
+     * settings cause the Printable.print() method to be called with the same page index more than once.
+     */
+    private static class TestMultiPagePrinterJob extends FakePrinterJob {
+        /*
+         * Print the document.
+         */
+        @Override
+        public void print() throws PrinterException {
+            // Create a BufferedImage to render into
+            final int width = (int) (format.getImageableWidth() - format.getImageableX());
+            final int height = (int) (format.getImageableHeight() - format.getImageableY());
+            // NOTE: We use a TYPE_4BYTE_ABGR because it is guaranteed to use a single contiguous
+            // image data buffer. This lets us checksum the raw data for the entire image.
+            final BufferedImage image = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+
+            Graphics2D gfx2d = image.createGraphics();
+
+            final int[] pageIndexes = { 0, 0, 1, 1 };
+
+            for (final int pageIndex : pageIndexes) {
```

Comment:
We can 'trick' it into saving all four pages (but still double call the print method) with some integer division:

```java
for (int i = 0; i < 4; i++) {
    if (painter.print(gfx2d, format, (i / 2)) == Printable.NO_SUCH_PAGE) {
    ...
    processPageImage(image, i);
    ...
}
```